samples-lib/src/main/java/com/esri/arcgismaps/sample/sampleslib/DownloaderActivity.kt

```diff
@@ -224,59 +231,53 @@ abstract class DownloaderActivity : AppCompatActivity() {
                 emit(LoadStatus.FailedToLoad(it))
                 return@flow
             }
+            // set up the file at the download path
+            val destinationFile = File(provisionLocation.path, portalItem.name)
+            // set up the download URL
+            val downloadURL =
+                "${portalItem.portal.url}/sharing/rest/content/items/${portalItem.itemId}/data"
             // get the data of the PortalItem
-            val portalItemData = portalItem.fetchData()
-            val byteArray = portalItemData.getOrElse {
-                // close dialog and emit status
-                loadingDialog.dismiss()
-                emit(LoadStatus.FailedToLoad(it))
-                return@flow
-            }
-            // get the byteArray of the PortalItem
-            runCatching {
-                val byteArrayInputStream = ByteArrayInputStream(byteArray)
-                val data = ByteArray(1024)
-                var downloadCount: Int
-                downloadCount = byteArrayInputStream.read(data)
-                while (downloadCount != -1) {
-                    downloadCount = byteArrayInputStream.read(data)
+            ArcGISEnvironment.arcGISHttpClient.download(
+                url = downloadURL,
+                destinationFile = destinationFile
+            ) { totalBytes, bytesRead ->
+                if (totalBytes != null) {
```

Comment:
if `totalBytes` is null, the ` progressIndicator` will be not shown, but downloading may still happen, since the server may not provide the property.  It is suggested to show a UX Spinner since we don't know how much in progress.

Reply:
maybe you could keep the old spinner to show if the `totalBytes` is null.